src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -181,13 +181,14 @@ impl<'a> JobQueue<'a> {
                     match result {
                         Ok(()) => self.finish(key, cx)?,
                         Err(e) => {
+                            cx.config.shell().error(format!("{}", e))?;
```

Comment:
Yep, I think you're right, that's better.
I've made some modifications in `cargo/lib.rs` to handle that.

The result is like this:
- First call is a normal `cargo build`
- Second call has the flag `--verbosity`

![screen shot 2017-01-25 at 19 59 01](https://cloud.githubusercontent.com/assets/2451004/22304770/1bccbe40-e339-11e6-9113-e54422e34ee9.png)